### ISSUE TITLE
"This combination of temperature and humidity sensors is already configured"

### ISSUE THREAD

**Issue by THX2112** [dolezsa/thermal_comfort#152]
Getting the error message "This combination of temperature and humidity sensors is already configured" when trying to add an additional room sensor.

There are no other Thermal Comfort instances using these sensors. I renamed those room sensors just to make sure (they were showing up as the default sensors when configuring Thermal Comfort.)

No error messages in the logs. Updated everything to the latest versions.

**Comment by rautesamtr** [comment]
Do you have any info on the input sensors? Do they have a unique_id? The unique id generated is normaly `temperature_sensor_unique_id`-`humidity_sensor_unique_id`. The error sounds like theres already something configured with the same unique_id.

**Comment by THX2112** [comment]
Where would I find this? The info page for the sensors just show the assigned name and entity ID. Nothing comes up at all when I search for "unique_id".

**Comment by SergioRius** [comment]
> Do you have any info on the input sensors? Do they have a unique_id? The unique id generated is normaly `temperature_sensor_unique_id`-`humidity_sensor_unique_id`. The error sounds like theres already something configured with the same unique_id.

Nope. I've realized that the identifier is generated with the friendly name. I have a ble integration that creates all the sensors with unique unique_ids, but uses simply temperature/hummidity for the friendly names. Those throw the error just at the 2nd added.

But using other sensors that had unique ids and friendly names, eventually also gave me that error.

**Comment by rautesamtr** [comment]
I wasn't able to reproduce this yet. Any simple setup of sensors to reproduce it?

**Comment by ovizii** [comment]
I got the same error:

`This combination of temperature and humidity sensors is already configured.`

![image](https://user-images.githubusercontent.com/837902/213557904-89ef08bd-9a64-4652-9f0f-b15ae27ee703.png)

![image](https://user-images.githubusercontent.com/837902/213558067-3da21f51-7640-4609-b805-99099df835d4.png)

I already added the first one as a TC integration. 

![image](https://user-images.githubusercontent.com/837902/213558183-d97275a2-235a-47ae-80ef-44b649312ca3.png)

Trying to set up the second TC integration:
![image](https://user-images.githubusercontent.com/837902/213558363-7bbaa3bd-676b-4fcc-9fe0-2c66b2e7f9d0.png)

![image](https://user-images.githubusercontent.com/837902/213558410-7b17c11d-dc45-4975-b129-1c4e0e4c244a.png)

**Comment by CEllwart** [comment]
I have the same error/problem here too.

**Comment by rautesamtr** [comment]
I added a debug message in regards of this error message.

Don't forget to add debug output to your configuration:
```yaml
logger:
  default: info
  logs:
…
    custom_components.thermal_comfort: debug
…
```

and look for the log message e.g.:
```
2023-02-07 12:43:09.204 DEBUG (MainThread) [custom_components.thermal_comfort.config_flow] An entry with the unique_id sensor_1-sensor_2 already exists: {'name': 'Outside', 'temperature_sensor': 'sensor.outside_temperature', 'humidity_sensor': 'sensor.outside_humidity', 'poll': False, 'scan_interval': 30, 'custom_icons': False}
```

So we can check if and which config entry already uses the unique_id.

**Comment by CEllwart** [comment]
`2023-02-07 17:46:22.534 DEBUG (MainThread) [custom_components.thermal_comfort.config_flow] An entry with the unique_id 0xa4c1384f2926c9f1_temperature_zigbee2mqtt-0xa4c1384f2926c9f1_humidity_zigbee2mqtt already exists: {'name': 'Küche - Comfort', 'temperature_sensor': 'sensor.0xa4c1384f2926c9f1_temperature', 'humidity_sensor': 'sensor.0xa4c1384f2926c9f1_humidity', 'poll': False, 'scan_interval': 240, 'custom_icons': False}`

The problem with this is:
that sensor i want to add was configured within "Küche - Comfort" in the past. But it was deconfigured and is no longer in use there (and not shown).

![grafik](https://user-images.githubusercontent.com/11425935/217310652-c3d2409b-a4b1-4223-80f3-6fc0cd2dc162.png)
![grafik](https://user-images.githubusercontent.com/11425935/217310766-d9b3f632-ffaf-44ed-8e53-6a240b26c494.png)

**Comment by rautesamtr** [comment]
I am unsure why home assistant keeps those old config entries. Could you check `config/.storage/core.config_entries`. You could try to delete the entry there, but make sure to stop home assistant before doing that and make a backup of the file.

**Comment by djwafelap** [comment]
I had the same message because I had reused two devices and solved it as follows:

A few months ago, I thought two of my thermometers were faulty, and replaced them with two new devices that were given the device name of the faulty thermometers. Later it turned out that the old thermometers were not defective, and I re-integrated them via zigbee2mqtt, but with a new device name (different room). These device were rejected by thermal comfort.

I played for a while with the unique_id's in the files /config/.storage/core.device_registry and /config/.storage/core.config_entries, but HA reversed my changes every time.

In the end, however, the solution was much simpler. I noticed that the entries under the old integration names had 2 devices.

![image](https://github.com/dolezsa/thermal_comfort/assets/125517688/0e9ef9b3-1f7b-4440-a9dd-3a0020cc95a7)

So I simply deleted these entries from thermal comfort and reinstalled them with the same name. The problem was gone.

I hope the solution works for you, too!

**Comment by 2fst4u** [comment]
I'm also getting this after I originally made the integration with two sensors (that I now want to use again) but I changed it to another pairing of sensors after creating. I now don't want to remove the working pair I have in case recreating both of them causes them both to give this error.

**Comment by Topherfx** [comment]
I'm getting the same message. I accidently deleted an integration entry und wanted to create it again with the same sensor. Something seems to be saved even though the Entry was deleted. Restarting HA a few times made no difference

**Comment by Topherfx** [comment]
Would be helpful to have a button "Forget old Configuration" or "create anyway" which deletes the old one. I don't feel competent enough to manually delete something from some config

**Comment by PiemurIsBack** [comment]
I have the same problem.
Is there any solution or a workaround to fix it?
I can´t edit core.config_entries.

**Comment by pdsccode** [comment]
Enabling debug log, trying to re-add the entry (which fails) and looking into the generated log led me to the solution. I moved a thermometer to another room as placed a different one with the same name in the original room. However, even after changing the sensors in TC, the debug log showed that the old thermometer was still used somehow. I then deleted the faulty TC config for the "old" room of the moved thermometer and re-added it with the new thermometer. After that, the entry for the moved one did work out.

**Comment by NovaLepidoptera** [comment]
I have the same problem. I had a look at this ticket, #401 and #408 as well. None of the proposed solutions worked in my case.
What I did: I created a thermal comfort in place a for sensor x, another one in place b for sensor y. Later on I moved x to b and y to place c (due to the sensors capabilities). I did delete the complete setup for place a, changed the ids for all the entities involved. Anyways I still get the error message "This combination of temperature and humidity sensors is already configured". In the logs I find the entry DEBUG (MainThread) [custom_components.thermal_comfort.config_flow] An entry with the unique_id 3E8737CB540D96F4-0000000000000016-MatterNodeDevice-1-TemperatureSensor-1026-0-3E8737CB540D96F4-0000000000000016-MatterNodeDevice-1-HumiditySensor-1029-0 already exists: {'custom_icons': False, 'humidity_sensor': 'sensor.a_x_humidity', 'name': 'Thermal Comfort', 'poll': False, 'scan_interval': 30, 'temperature_sensor': 'sensor.a_x_temperature'}. This entry uses the sensor names a_x_humidity and a_x_temperature even though I changed those to b_x_humidity and b_x_temperature.
Additionally, I deleted the statistics as well. The only thing I did not yet try, was fiddling with with the unique_id's in the files /config/.storage/core.device_registry and /config/.storage/core.config_entries.

Any ideas on what else I could do?
